src/web/app/views/beambox/Right-Panels/ConfigPanel/DottingTimeBlock.tsx

```diff
@@ -0,0 +1,86 @@
+import classNames from 'classnames';
+import React, { memo, useContext, useMemo } from 'react';
+
+import eventEmitterFactory from 'helpers/eventEmitterFactory';
+import history from 'app/svgedit/history/history';
+import ISVGCanvas from 'interfaces/ISVGCanvas';
+import ObjectPanelItem from 'app/views/beambox/Right-Panels/ObjectPanelItem';
+import UnitInput from 'app/widgets/Unit-Input-v2';
+import useI18n from 'helpers/useI18n';
+import { CUSTOM_PRESET_CONSTANT, writeData } from 'helpers/layer/layer-config-helper';
+import { getSVGAsync } from 'helpers/svg-editor-helper';
+import { LayerPanelContext } from 'app/views/beambox/Right-Panels/contexts/LayerPanelContext';
+
+import ConfigPanelContext from './ConfigPanelContext';
+import styles from './Block.module.scss';
+
+let svgCanvas: ISVGCanvas;
+getSVGAsync((globalSVG) => {
+  svgCanvas = globalSVG.Canvas;
+});
+
+const DottingTimeBlock = ({
+  type = 'default',
+}: {
+  type?: 'default' | 'panel-item' | 'modal';
+}): JSX.Element => {
+  const lang = useI18n();
+  const t = lang.beambox.right_panel.laser_panel;
+
+  const { selectedLayers, state, dispatch, initState } = useContext(ConfigPanelContext);
+  const { dottingTime } = state;
+  const timeEstimationButtonEventEmitter = useMemo(
+    () => eventEmitterFactory.createEventEmitter('time-estimation-button'),
+    []
+  );
+  const { hasGradient } = useContext(LayerPanelContext);
+  if (!hasGradient) return null;
+
+  const handleChange = (value: number) => {
+    dispatch({
+      type: 'change',
+      payload: { dottingTime: value, configName: CUSTOM_PRESET_CONSTANT },
+    });
+    timeEstimationButtonEventEmitter.emit('SET_ESTIMATED_TIME', null);
+    if (type !== 'modal') {
+      const batchCmd = new history.BatchCommand('Change dotting time');
+      selectedLayers.forEach((layerName) => {
+        writeData(layerName, 'dottingTime', value, { batchCmd });
+        writeData(layerName, 'configName', CUSTOM_PRESET_CONSTANT, { batchCmd });
```

Comment:
do we need to change preset name if dotting time changed?

Reply:
Yes